### PR TITLE
Fix Demo Build Errors

### PR DESCRIPTION
**Pull request by Skptak**
Description
-----------

Shrink the idle task stack and the timer callback stack by 512, and reduce the number of OTA buffers held to fit into minimal size when using ESP IDF v5.1

Test Steps
-----------

Shrank sizes using `idf.py menuconfig`, copied relevant config options to the default config.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

#49 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by paulbartell — DISMISSED** [review]

**Review by rawalexe — APPROVED** [review]

File: sdkconfig.defaults [hunk 1]
```diff
@@ -24,3 +25,33 @@ CONFIG_ESP_SECURE_CERT_SUPPORT_LEGACY_FORMATS=y
...
+
+# AWS OTA
+CONFIG_LOG2_FILE_BLOCK_SIZE=12
+CONFIG_MAX_NUM_BLOCKS_REQUEST=8
+CONFIG_MAX_NUM_OTA_DATA_BUFFERS=8
+CONFIG_ALLOW_DOWNGRADE=0
+CONFIG_OTA_DATA_OVER_MQTT=y
+# CONFIG_OTA_DATA_OVER_HTTP is not set
+CONFIG_OTA_DATA_OVER_MQTT_PRIMARY=y
+CONFIG_OTA_PRIMARY_DATA_PROTOCOL=1
```
Comment (rawalexe) [inline sdkconfig.defaults:47]: Just out of concern, Where are you getting these numbers from?

File: sdkconfig.defaults [hunk 2]
```diff
@@ -24,3 +25,33 @@ CONFIG_ESP_SECURE_CERT_SUPPORT_LEGACY_FORMATS=y
...
 CONFIG_GRI_SUB_PUB_UNSUB_DEMO_DELAY_BETWEEN_SUB_PUB_UNSUB_LOOPS_MS=10000
 CONFIG_GRI_MQTT_AGENT_CONNACK_RECV_TIMEOUT_MS=10000
+
+# Set optimization for size
+CONFIG_COMPILER_OPTIMIZATION_SIZE=y
+
+# Set default FreeRTOS Stack sizes
+CONFIG_FREERTOS_TIMER_TASK_STACK_DEPTH=1536
+CONFIG_FREERTOS_IDLE_TASK_STACKSIZE=1024
+
```
Comment (rawalexe) [inline sdkconfig.defaults:35]: These too

File: GettingStartedGuide.md [hunk 3]
```diff
@@ -84,16 +122,31 @@ The demo will connect to the AWS IoT device Endpoint that you configure here.
...
+  - **NOTE:** If using the
+    [AWS IoT Generated Credentials](https://docs.aws.amazon.com/iot/latest/developerguide/device-certs-create.html)
+    this value will be `rsa 2048`
+
+**NOTE** The list of chip types can be found by running `idf.py list-targets`
+The directly supported chips are the `esp32c3`, `esp32s3`, and `esp32c2`
+
 ```sh
-python managed_components/espressif__esp_secure_cert_mgr/tools/configure_esp_secure_cert.py -p PORT --keep_ds_data_on_host --ca-cert CA_CERT_FILEPATH --device-cert DEVICE_CERT_FILEPATH --private-key PRIVATE_KEY_FILEPATH --target_chip esp32c3 --secure_cert_type cust_flash
+idf.py set-target CHIP_TYPE;
```
Comment (xuelix) [inline GettingStartedGuide.md:146]: set-target should be done first, otherwise, endpoint and device name have to be set again.

**Review by xuelix — CHANGES_REQUESTED** [review]

File: GettingStartedGuide.md (hunk 3 above)
Comment (xuelix) [reply GettingStartedGuide.md:146]: This line should be moved to 2.2, before step 1, before running idf.py menuconfig

File: sdkconfig.defaults (hunk 1 above)
Comment (Skptak) [reply sdkconfig.defaults:47]: I got them by running `idf.py menuconfig` changing the settings a bit to shrink the image, and then copying them from the `sdkconfig` that gets generated

File: sdkconfig.defaults (hunk 2 above)
Comment (Skptak) [reply sdkconfig.defaults:35]: I got them by running `idf.py menuconfig` and selecting the minimal supported task stack size for these features

I ran the demos for a bit and didn't see the stack overflow check get raised, so don't believe the sizes are needed. The previous defaults were `1536` for the idle task and `2048` for the timer task.

File: GettingStartedGuide.md [hunk 4]
```diff
@@ -84,16 +122,31 @@ The demo will connect to the AWS IoT device Endpoint that you configure here.
...
+- `DEVICE_CERT_FILEPATH`: The file path to the PEM-encoded device certificate
+
+- `PRIVATE_KEY_FILEPATH`: The file path to the PEM-encoded private key.
+
+- `KEY_ALG_INFO`: The type of key algorithm being used. In the
+format Algorithm Size,
+
+  - **NOTE:** If using the
+    [AWS IoT Generated Credentials](https://docs.aws.amazon.com/iot/latest/developerguide/device-certs-create.html)
+    this value will be `rsa 2048`
```
[the remark this answers is not in the document]
Comment (Skptak) [reply GettingStartedGuide.md:140]: I was able to do it with lower case, I think it takes it either way
I prefer the capital one though, so if that worked for you I can change it

File: GettingStartedGuide.md (hunk 3 above)
Comment (Skptak) [reply GettingStartedGuide.md:146]: Oops, thought I had it before then, thanks for noticing!

File: GettingStartedGuide.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (xuelix) [reply GettingStartedGuide.md:140]: lower case is not working for me.

[the remark this answers is not in the document]
Comment (Skptak) [reply GettingStartedGuide.md]: I swapped it to uppercase, maybe it's an OSX thing

[the remark this answers is not in the document]
Comment (Skptak) [reply GettingStartedGuide.md]: https://github.com/FreeRTOS/iot-reference-esp32c3/pull/68/commits/7d6428db03660cc74eb1b3a04a6700e872e0227f#diff-7eb4ec8845a00e67a78ed3ef84df2e03815d85424965865b0bb1c95d9b23bb2eR147

**Review by xuelix — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**